tests/TestCase/Collection/CollectionTest.php

```diff
@@ -1007,16 +1007,29 @@ public function testInvalidConstructorArgument()
     }
 
     /**
-     * Tests that issuing a count will throw an exception
+     * Tests that Count returns the number of elements
      *
+     * @dataProvider simpleProvider
      * @return void
      */
-    public function testCollectionCount()
+    public function testCollectionCount($list)
     {
-        $this->expectException(\LogicException::class);
-        $data = [1, 2, 3, 4];
-        $collection = new Collection($data);
-        $collection->count();
+        $list = (new Collection($list))->buffered();
+        $collection = new Collection($list);
+        $this->assertEquals(8, $collection->append($list)->count());
```

Comment:
Should there be tests for any of the sharp edges this method has?

Reply:
any case you arrnintested in? The current test shows how a collection with only 4 unique keys can have 8 elements. That odd case is covered

Reply:
Good point. On second thought the other cases are likely not worth covering with tests.